Review of the hostname topic (#9952)
keycloak/keycloak: docs/guides/src/main/server/hostname.adoc

summary="Learn how to configure the frontend and backchannel endpoints exposed by Keycloak."
includedOptions="hostname-* proxy">

When running Keycloak in environments such as Kubernetes, OpenShift, or on-premise, you want to protect the internal URLs from exposure to the public facing internet. You want to use your public hostname. This guide describes how to configure Keycloak to use the right hostname for different scenarios.

== Keycloak API Endpoint categories
As this section explains, Keycloak exposes three API endpoint categories: frontend, backend, and administrative.  Each category uses a specific base URL. 

=== Frontend Endpoints
Frontend endpoints are used to externally access Keycloak. When no hostname is set, the base URL used for the frontend is taken from the incoming request. This choice has some major disadvantages. For example, in a high availability scenario, you may have multiple Keycloak instances. The choice of URL should not depend on the instance where the request lands. The URL should be used for all instances, so they are seen as one system from the outside.

> To set the hostname part of the frontend base URL, enter the `kc.start` command in this format:

Reviewer: `kc.start` is not a command, but rather a FreeMarker macro.
```suggestion
To set the hostname part of the frontend base URL, enter this command:
```

<@kc.start parameters="--hostname=<value>"/>

=== Backend Endpoints

> Backend endpoints are used for direct communication between Keycloak and applications. Examples of backend endpoints are the Token endpoint and the User info endpoint. Backend endpoints are also taking the base URL from the request by default. To override this behavior, set the `hostname-strict-backchannel` configuration option by entering the `kc.start` command in this format:

Reviewer: ```suggestion
Backend endpoints are used for direct communication between Keycloak and applications. Examples of backend endpoints are the Token endpoint and the User info endpoint. Backend endpoints are also taking the base URL from the request by default. To override this behavior, set the `hostname-strict-backchannel` configuration option by entering this command:
```

<@kc.start parameters="--hostname=<value> --hostname-strict-backchannel=true"/>

When all applications connected to Keycloak communicate through the public URL, set `hostname-strict-backchannel` to true. Otherwise, leave this parameter as false to allow internal applications to communicate with Keycloak through an internal URL.

=== Administrative Endpoints

> When the Admin Console is exposed on a different hostname, use `--hostname-admin` to link to it.  Enter the `kc.start` command in this format:

Reviewer: ```suggestion
When the Admin Console is exposed on a different hostname, use `--hostname-admin` to link to it, as shown in this example:
```
